manager.go

```
@@ -579,10 +580,65 @@ func (em *Manager) RemoveListeners(name string) {
+	events := sbr.SubscribedEvents()
+	for name, listener := range events {
+		switch lt := listener.(type) {
+		case Listener[T], *Listener[T], ListenerFunc[T], *ListenerFunc[T]:
+			// Directly compatible types
+			var l Listener[T]
+			switch v := lt.(type) {
+			case Listener[T]:
+				l = v
+			case *Listener[T]:
+				l = *v
+			case ListenerFunc[T]:
```

golangci-lint (1.64.8)

SA4020: unreachable case clause: go.lumeweb.com/event.Listener[T] will always match before go.lumeweb.com/event.ListenerFunc[T]

(staticcheck)